### ISSUE TITLE
how to add more conv layers to demo_conv_psd_gpu.lua?

### ISSUE THREAD

**Issue by kingofoz** [viorik/unsupgpu#5]
Hi viorik,
If I want to get a better performance, I should add more conv layers, right? Can you show me how to do this to encoder and decoder?

Thanks,
Yingjun

**Comment by viorik** [comment]
Hi @kingofoz 
What exactly do you mean by better performance? 
I haven't tried to train a deeper autoencoder with this framework. I would be curious about the results though, first of all if the training converges at all. To add layers, you would probably need to add extra SpatialConvolution+Tanh layers to the encoder in demo_conv_psd_gpu.lua, at line 114. Also, deeper architectures usually use pooling+subsampling between convolutional layers, so you should add that as well. Then for the decoder, you need to modify SpatialConvFistaL1, declare D as nn.Sequential, and add convolutional layers, symmetric to the encoder (with upsampling layers if you used pooling+downsampling in the encoder).
Good luck! Let me know if you manage to train it.

**Comment by kingofoz** [comment]
Hi viorik,
Thanks for the info! I trained a model yesterday night. The final average loss is about 3.9, I think maybe I can add more conv layer to get a lower loss?
Another problem is the training stuck at 995001 iteration. No response, I have to enter Ctrl + C to terminate it. Please see the below final output:
==> iteration = 995001, average loss = 3.9805686359346
 [============================================================== 4995/5000 ===========>.] ETA: 208ms | Step: 41ms  
^C

By the way, is average loss the test error? I am very new to Torch and autoencoder, sometimes I ask stupid questions. :-)

Thanks,
Yingjun

**Comment by viorik** [comment]
Hi @kingofoz
This is already a design general discussion. I suggest you to close this issue and send me your email address and we can continue the discussion there. My email is vpatrauc at gmail dot com.

**Comment by kingofoz** [comment]
Sure. Let me close it now.

**Comment by kingofoz** [comment]
Please see my email, I reported a new issue about kernelsize. @viorik
